engagement_database/engagement_database.py

```diff
@@ -275,6 +275,37 @@ def delete_doc(self, path, transaction=None):
         else:
             transaction.delete(ref)
 
+    def delete_message_and_history(self, message_id, transaction=None):
+        """
+        Deletes a message and its history entries.
+
+        :param message_id: Id of message to delete.
+        :type message_id: str
+        :param transaction: Transaction to run this update in or None.
+                            If None, writes immediately, otherwise adds the updates to a transaction that will need
+                            to be explicitly committed elsewhere.
+        :type transaction: google.cloud.firestore.Transaction | None
+        """
+        message_snapshot = self._message_ref(message_id).get(transaction=transaction)
```

Comment:
```suggestion
        message = self.get_message(message_id, transaction)
```

(renaming to message for consistency with history_entries on the next line, which doesn't mention snapshots)